clients/agentic-sandbox-client-python/agentic_sandbox/sandbox.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2025 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+import subprocess
+from dataclasses import dataclass
+
+import requests
+from kubernetes import client, config, watch
+
+# Constants for SandboxClaim
+CLAIM_API_GROUP = "extensions.agents.x-k8s.io"
+CLAIM_API_VERSION = "v1alpha1"
+CLAIM_PLURAL_NAME = "sandboxclaims"
+
+# Constants for Sandbox
+SANDBOX_API_GROUP = "agents.x-k8s.io"
+SANDBOX_API_VERSION = "v1alpha1"
+SANDBOX_PLURAL_NAME = "sandboxes"
+
+@dataclass
+class ExecutionResult:
+    """A structured object for holding the result of a command execution."""
+    stdout: str
+    stderr: str
+    exit_code: int
+
+class Sandbox:
+    """
+    The main client for creating and interacting with a stateful Sandbox.
+    This class is a context manager, designed to be used with a `with` statement.
+    """
+    def __init__(self, template_name: str, namespace: str = "default"):
+        self.template_name = template_name
+        self.namespace = namespace
+        self.claim_name: str | None = None
+        self.sandbox_name: str | None = None
```

Comment:
The `server_port` is hardcoded to `8888`. It would be more flexible if this was a parameter to the `__init__` method with a default value. This would allow users to override the port if needed.